### PR TITLE
Fixed wxwidgets

### PR DESCRIPTION
**Pull request by ioandev**
I've successfully compiled wxwidgets and tested it on a yocto machine

### CODE REVIEW

File: recipes-extended/wxwidgets/wxwidgets_3.0.3.1.bb [hunk 1]
```diff
@@ -4,45 +4,37 @@ HOMEPAGE = "http://www.wxwidgets.org"
...
+#inherit autotools
+inherit autotools pkgconfig
 
+EXTRA_OECONF = "  --disable-shared \
+	--enable-unicode \
+	"
 CXXFLAGS := "${@oe_filter_out('-fvisibility-inlines-hidden', '${CXXFLAGS}', d)}"
 CXXFLAGS += "-std=gnu++11"
 
-# Broken autotools :/
```
Comment (bulwahn) [inline recipes-extended/wxwidgets/wxwidgets_3.0.3.1.bb:30]: If autotools is not broken anymore, can we simply drop the special `do_configure` in this recipe?

File: recipes-extended/wxwidgets/wxwidgets_3.0.3.1.bb [hunk 2]
```diff
@@ -4,45 +4,37 @@ HOMEPAGE = "http://www.wxwidgets.org"
...
-EXTRA_AUTORECONF = " -I ${S}/build/aclocal"
-EXTRA_OECONF = "  --with-opengl \
-                 --without-sdl \
-                 --enable-mediactrl=yes \
-                 --enable-webviewwebkit=yes \
-                 --disable-gpe \
-                 --disable-visibility \
-                 --disable-rpath \
-               "
+#inherit autotools pkgconfig binconfig
```
Comment (bulwahn) [inline recipes-extended/wxwidgets/wxwidgets_3.0.3.1.bb:15]: the commented-out lines can just be dropped; it looks like some left-overs from your development and testing.

File: recipes-extended/wxwidgets/wxwidgets_3.0.3.1.bb [hunk 3]
```diff
@@ -4,45 +4,37 @@ HOMEPAGE = "http://www.wxwidgets.org"
...
 CXXFLAGS += "-std=gnu++11"
 
-# Broken autotools :/
 do_configure() {
 	oe_runconf
 }
 
 # wx-config contains entries like this:
 # this_prefix=`check_dirname "/build/v2013.06/build/tmp-angstrom_v2013_06-eglibc/work/cortexa8hf-vfp-neon-angstrom-linux-gnueabi/wxwidgets/2.9.5-r0/wxWidgets-2.9.5"`
 do_install_prepend() {
```
Comment (bulwahn) [inline recipes-extended/wxwidgets/wxwidgets_3.0.3.1.bb:31]: An empty `do_install_prepend` and its comment can just be dropped if it is not needed anymore.

File: recipes-extended/wxwidgets/wxwidgets_3.0.3.1.bb [hunk 4]
```diff
@@ -4,45 +4,37 @@ HOMEPAGE = "http://www.wxwidgets.org"
...
-	sed -i -e s:${S}:${STAGING_DIR_HOST}${prefix}:g ${S}/wx-config
+
 }
 
 # wx-config doesn't handle the suffixed libwx_media, xrc, etc, make a compat symlink
 do_install_append() {
```
Comment (bulwahn) [inline recipes-extended/wxwidgets/wxwidgets_3.0.3.1.bb:36]: An empty `do_install_append` can be dropped as well.

**Comment by bulwahn** [comment]
I just provided a few comments for the review; other than that it looks good. I will need to test that later this week.

**Comment by ioandev** [comment]
Hi,

I agree with everything you said, there is some left over stuff because I only wanted to get it working.

I am trying to get wxpython working, but I am just starting with yocto so I don't understand a possibly simple thing.
wxpython needs wxwigets' include header files, but wxwidgets deletes them after it's compiled. How do I make wxwidgets keep them there a while longer? I think I got it into such a state and it yelled at me for not deleting the left over files and actually errored out.
Or does it not delete the include files, but I am unaware where the include files are? I am expecting them to be /usr/include.

Regards,
Ioan

**Comment by ioandev** [comment]
I've changed both wxwidgets and wxpython again to accommodate each other. Unfortunately after testing the build i find out the wx module is not present in python.

Is it possible I can run python setup.py install ? I can't find a good example recipe for this. This step seems to be part of the build as said in https://wiki.wxpython.org/How%20to%20install%20wxPython

**Comment by ioandev** [comment]
To recap: both now build, but only wxwidgets is visible on the device.

**Comment by bulwahn** [comment]
@ioanb7 You can achieve this, e.g., with a special`do_install_append` command in the recipe. Alternatively, you could `inherit distutils`, which should call this correctly for the cross-compiliation setup already as part of its `do_install`. Just have a look at the distutils.bbclass in openembedded-core.

**Comment by ioandev** [comment]
It worked, thanks so much. 

I do have another error now:
root@imx6ul-var-dart:/usr/bin# pip freeze
smart==1.5
wxPython==3.0.2.0
wxPython-common==3.0.2.0
You are using pip version 8.1.2, however version 9.0.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
root@imx6ul-var-dart:/usr/bin# python
Python 2.7.12 (default, Sep  2 2017, 23:59:37) 
[GCC 6.2.0] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import wx
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python2.7/site-packages/wx-3.0-gtk2/wx/__init__.py", line 45, in <module>
    from wx._core import *
  File "/usr/lib/python2.7/site-packages/wx-3.0-gtk2/wx/_core.py", line 4, in <module>
    import _core_
ImportError: /usr/lib/python2.7/site-packages/wx-3.0-gtk2/wx/_core_.so: undefined symbol: _ZTI12wxEvtHandler

But I think it's because i gave it gtk3 instead of gtk2.

**Comment by bulwahn** [comment]
Sorry, this question is too difficult for me to answer without further own investigation and reading of build scripts and code. Unfortunately, I also have too many other things on my todo list; so, I probably cannot help you with this. I will certainly test and merge your pull request, once you consider the pull request ready for review and merge.

**Comment by bulwahn** [comment]
Is there any progress on this pull request? What is the currently blocking issue?

**Comment by ioandev** [comment]
Hi, although this was building it didn't work after installing. I was able to import the package in wxpython but it gave an error quickly afterwards.

**Comment by bulwahn** [comment]
Is the current pull request the current state that you have? I might have an hour or two to look at it in the next few days.

**Comment by ioandev** [comment]
I am going to try and find the wxpython bb file.
You can then compile both of them and go inside python and do `import wx` and you should get the error i was getting.
I've moved to gtkmm as this didn't work and I was getting no where.

**Comment by bulwahn** [comment]
Okay, should we just close this pull request then?

**Comment by ioandev** [comment]
It depends if you want to give it a try. It might work for you or at least you might find the issue.

Let me know

**Comment by ioandev** [comment]
Sadly I am unable to find the bb for wxpython. I must've lost the backup.

Closing this for now.

**Comment by bulwahn** [comment]
That is really unfortunate that you lost that backup. Just let us know if you find it again or we re-tackle the wx/wxpython problem.

**Pull request closed without merging**